src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/RoundTripIslSpec.groovy

```diff
@@ -0,0 +1,287 @@
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.DISCOVERED
+
+        when: "Deactivate the dst switch"
+        lockKeeper.knockoutSwitch(dstSwToDeactivate)
+        def isDstSwDeactivated = true
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(dstSwToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "The round trip latency ISL is FAILED (because round_trip_status is not available in DB for current ISL \
+on both switches)"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.FAILED
+
+        cleanup:
+        isSrcSwDeactivated && lockKeeper.reviveSwitch(srcSwToDeactivate)
+        isDstSwDeactivated && lockKeeper.reviveSwitch(dstSwToDeactivate)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getSwitch(dstSwToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.DISCOVERED
+            }.size() == topology.islsForActiveSwitches.size() * 2
+        }
+    }
+
+    @Tidy
+    def "A round trip latency ISL goes down when the src switch is DEACTIVATED and round trip latency rule is removed \
+on the dst switch"() {
+        given: "A round trip latency ISL"
+        Isl roundTripIsl
+        def srcSwToDeactivate = topology.activeSwitches.find { sw ->
+            if (sw.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)) {
+                roundTripIsl = topology.getRelatedIsls(sw).find {
+                    it.dstSwitch.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)
+                }
+                roundTripIsl
+            }
+        } ?: assumeTrue("Wasn't able to find a switch with suitable link", false)
+        def dstSw = roundTripIsl.dstSwitch
+
+        and: "Round trip status is ACTIVE for the given ISL in both directions"
+        [roundTripIsl, roundTripIsl.reversed].each { isl ->
+            assert database.getIslRoundTripStatus(isl) == IslStatus.ACTIVE
+        }
+
+        when: "Deactivate the src switch"
+        lockKeeper.knockoutSwitch(srcSwToDeactivate)
+        def isSrcSwDeactivated = true
+        // it takes more time to DEACTIVATE a switch via the 'knockoutSwitch' method on the stage env
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "The round trip ISL is still DISCOVERED (the system uses round trip latency for ISL alive confirmation)"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
```

Comment:
Is it mandatory to verify this again and again? It is actually a very time-consuming verification

Reply:
fixed